File: python/chapter-2/hello_world_producer.py
###############################################
# RabbitMQ in Action
# Chapter 1 - Hello World Producer
# 
# Requires: pika >= 0.9.5
# 
# Author: Jason J. W. Williams
# (C)2011
###############################################

import pika, sys

credentials = pika.PlainCredentials("guest", "guest")
conn_params = pika.ConnectionParameters("localhost",
                                        credentials = credentials)
conn_broker = pika.BlockingConnection(conn_params) #/(hwp.1) Establish connection to broker


channel = conn_broker.channel() #/(hwp.2) Obtain channel

channel.exchange_declare(exchange="hello-exchange", #/(hwp.3) Declare the exchange
                         type="direct",
                         passive=False,
                         durable=True,
                         auto_delete=False)

msg = sys.argv[1]
msg_props = pika.BasicProperties()
msg_props.content_type = "text/plain" #/(hwp.4) Create a plaintext message

channel.basic_publish(body=msg,
                      exchange="hello-exchange",
                      properties=msg_props,
                      routing_key="hola") #/(hwp.5) Publish the message

File: ruby/chapter-2/hello_world_producer.rb
if defined?(Bundler); Bundler.setup; else; require "rubygems"; end

require "amqp"

EventMachine.run do
  AMQP.connect(:host => "localhost", :username => "guest", :password => "guest") do |connection|
    puts "Connected. Hit Control + C to stop me."

    channel  = AMQP::Channel.new(connection)
    channel.direct("hello-exchange", :durable => true, :auto_delete => false).
      publish(ARGV[0], :routing_key => "hola")

    # stop on Control + c or after 1 second
    stop = Proc.new { connection.close { EventMachine.stop } }
    Signal.trap("INT", &stop)
    EventMachine.add_timer(1, &stop)
  end
end